scripts/proto_codegen.sh

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+if ! [[ "$0" =~ scripts/proto_codegen.sh ]]; then
+  echo "must be run from repository root"
+  exit 255
+fi
+
+PROTOC_VERSION="27.1"
+if [[ $(protoc --version | cut -f2 -d' ') != "${PROTOC_VERSION}" ]]; then
+  # e.g., protoc 27.1
+  echo "could not find protoc ${PROTOC_VERSION}, is it installed + in PATH?"
+  exit 255
+fi
+
+PROTOC_GEN_GO_VERSION="v1.28.0"
+go install -v google.golang.org/protobuf/cmd/protoc-gen-go@${PROTOC_GEN_GO_VERSION}
+if [[ $(protoc-gen-go --version | cut -f2 -d' ') != "${PROTOC_GEN_GO_VERSION}" ]]; then
+  # e.g., protoc-gen-go v1.28.0
+  echo "could not find protoc-gen-go ${PROTOC_GEN_GO_VERSION}, is it installed + in PATH?"
+  exit 255
+fi
+
+PROTOC_GEN_GO_GRPC_VERSION="1.1.0"
+go install -v google.golang.org/grpc/cmd/protoc-gen-go-grpc@v${PROTOC_GEN_GO_GRPC_VERSION}
+if [[ $(protoc-gen-go-grpc --version | cut -f2 -d' ') != "${PROTOC_GEN_GO_GRPC_VERSION}" ]]; then
+  # e.g., protoc-gen-go-grpc 1.1.0
+  echo "could not find protoc-gen-go-grpc ${PROTOC_GEN_GO_GRPC_VERSION}, is it installed + in PATH?"
+  exit 255
+fi
+
+TARGET=$PWD/proto
+if [ -n "${1:-}" ]; then
+  TARGET="$1"
+fi
+
+cd "$TARGET"
+
+echo "Creating .pb.go files"
+protoc --go_out=. --go-grpc_out=. subscriber.proto
```

Comment:
Can we copy the same file structure from AvalancheGo https://github.com/ava-labs/avalanchego/tree/master/proto/pb ie. `proto/servicename/servicename.proto` and then putting the generated files in `proto/pb/servicename/...`

We should also copy the same style as used here: https://github.com/ava-labs/avalanchego/blob/master/scripts/protobuf_codegen.sh and make sure that we are using either the same versions as in AvalancheGo or the latest versions of each dependency.